src/windowmanagement/TaskBar.tsx

```diff
@@ -15,29 +16,51 @@ export function TaskBar({ addWindow, visibleMenuItems }: TaskBarProps) {
   });
 
   return (
-    <Group
+    <Flex
+      gap={2}
       style={{
         backgroundColor: '#00000080',
-        height: 60,
-        paddingLeft: 'var(--mantine-spacing-md)'
+        whiteSpace: 'nowrap',
+        overflowX: 'auto',
+        overflowY: 'hidden'
+      }}
+      onWheel={(event) => {
+        event.currentTarget.scrollLeft += event.deltaY;
       }}
     >
       {checkedMenuItems.map((item) => {
-        const handleClick = () =>
+        function handleClick() {
           addWindow(item.content, {
             title: item.title,
             position: item.preferredPosition,
             id: item.componentID
           });
+        }
 
-        return item.renderMenuButton ? (
-          item.renderMenuButton(item.componentID, handleClick)
-        ) : (
-          <Button key={item.componentID} size={'xl'} onClick={handleClick}>
-            {item.title}
-          </Button>
+        return (
+          <div style={{ flex: '0 0 auto' }}>
```

Comment:
You need a key here (the outermost element needs the key in a map)

Reply:
Also I was thinking whether we want to use `<div>` or `<Box>` from mantine. Box could be nice as it will be uniform and we can use mantine styling variables